astropy/io/fits/scripts/fitsdiff.py

```diff
@@ -232,7 +232,19 @@ def match_files(paths):
                 sys.exit(2)
             filelists.append(files)
         elif os.path.isdir(path):
-            filelists.append([os.path.abspath(f) for f in os.listdir(path)])
+            # Determine if the directory is in the first argument or second argument,
+            # select all files in the former case or
+            # files matching with the file name of first argument in the later case.
+            if i == 0:
+                filelists.append([os.path.abspath(os.path.join(path, f)) for f in os.listdir(path)])
+            else:
+                filelists.append([])
+                for f in filelists[0]:
+                    f_n = os.path.basename(f)
+                    p_n = os.path.join(path, f_n)
+                    if os.path.isfile(p_n):
+                        filelists[1].append(os.path.abspath(p_n))
```

Comment:
Thanks, looks better now. But in terms of behavior I think we need more, to quote the docstring at the top of file:

> If both are directory names, all files in each of
 the directories will be included; if only one is a directory name, then the
 directory name will be prefixed to the file name(s) specified by the other
argument.

So it should work if the first argument in a directory and the second a file or glob pattern. Thinking more about it, I think in this we should always all the files in the directory (so just `filelists.append([os.path.abspath(os.path.join(path, f)) for f in os.listdir(path)])` without the if/else, and then below rework the loop the files that are matching in both file lists. Does it make sense ?

Reply:
yes!it does make sense..i did not read the docstring.I just guessed it from the discussions on the issue.Do not worry! i will change it soon.!